### ISSUE TITLE
Deriving-via for HasHeader instance for blocks (to refer to the header)

### ISSUE THREAD

**Issue by edsko** [input-output-hk/ouroboros-network#1492]
We have two `HasHeader` instances for lots of blocks: one for the full block, and one for the header, where the former just delegates to the latter. It ought to be possible to use deriving-via for this.

**Comment by mrBliss** [comment]
No longer needed after #2329.
